### PR TITLE
Build PowerShell native assemblies and Publish NuGet package

### PR DESCRIPTION
**Pull request by adityapatwardhan**
* Create docker files for docker based build.
* Update build.psm1 to support latest SDK and Visual studio 2017.
* Create infrastructure for signing files.
* Create functions for generating a NuGet package.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -232,8 +242,18 @@ function Start-BuildNativeWindowsBinaries {
     Write-Log "Start building native Windows binaries"
 
     if ($Clean) {
-        git clean -fdx
-        Remove-Item $HOME\source\cmakecache.txt -ErrorAction SilentlyContinue
+        Write-Verbose "Pushing location: $(Get-Location)" -Verbose
```
Comment (TravisEz13) [inline build.psm1]: Use Write-Log

File: build.psm1 [hunk 2]
```diff
@@ -398,81 +437,224 @@ function Start-BuildNativeUnixBinaries {
...
+    $RuntimePathLinux = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/linux-x64/native') -Force
+    $RuntimePathLinuxARM = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/linux-arm/native') -Force
+    $RuntimePathOSX = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/osx/native') -Force
+
+    Copy-Item "$BinFolderLinux\*" -Destination $RuntimePathLinux -Verbose
+    Copy-Item "$BinFolderLinuxARM\*" -Destination $RuntimePathLinuxARM -Verbose
+    Copy-Item "$BinFolderOSX\*" -Destination $RuntimePathOSX -Verbose
+}
+
+function PlaceWindowsNativeBinaries
```
Comment (TravisEz13) [inline build.psm1]: you should add some form of header... at least a synopsis

File: tools/releaseBuild/build.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            "AdditionalContextFiles": [
+                "./tools/releaseBuild/PowershellNative.ps1"
+            ],
+            "DockerFile": "./tools/releaseBuild/images/CentOS/Dockerfile",
+            "DockerImageName": "ps-centos-7",
+            "BinaryBucket": "release",
+            "EnableFeature": [ "ArtifactAsFolder" ]
+        }
+    ]
+}
```
Comment (TravisEz13) [inline tools/releaseBuild/build.json]: missing newline

Comment (TravisEz13) [reply tools/releaseBuild/build.json]: in a few files

File: tools/releaseBuild/signing.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<SignConfigXML>
+    <!-- ****Begin**** Authenticode - Authenticode SHA256 ******** -->
+    <job platform="" configuration="" dest="__OUTPATHROOT__\signed" jobname="PowerShell Core" approvers="vigarg;gstolt">
+        <file src="__INPATHROOT__\PowerShell.Core.Instrumentation.dll" signType="Authenticode" dest="__OUTPATHROOT__\PowerShell.Core.Instrumentation.dll" />
```
Comment (TravisEz13) [inline tools/releaseBuild/signing.xml]: you don't need the update signing step in your build with this xml... try disabling it.

File: tools/releaseBuild/vstsBuild.ps1 [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    return $Dict
+}
+
+begin {
+    $Name = $PSBoundParameters['Name']
+}
+
+end {
+
+    $psReleaseBranch = 'master'
```
Comment (TravisEz13) [inline tools/releaseBuild/vstsBuild.ps1]: Do you want to create a branch so you can control when you take changes?

File: tools/releaseBuild/signing.xml (hunk 4 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/signing.xml]: Tried. Does not work.

File: tools/releaseBuild/vstsBuild.ps1 (hunk 5 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/vstsBuild.ps1]: I am fine with keeping this as master.

File: build.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply build.psm1]: Fixed.

File: build.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply build.psm1]: Added a synopsis for all new functions

File: tools/releaseBuild/build.json (hunk 3 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/build.json]: Add all new missing newlines

**Comment by adityapatwardhan** [comment]
@TravisEz13  Please have another look.

File: tools/releaseBuild/signing.xml (hunk 4 above)
Comment (TravisEz13) [reply tools/releaseBuild/signing.xml]: thanks for trying

**Pull request merged**